app/labor/rate_limit_checker.rb

```diff
@@ -1,5 +1,6 @@
 class RateLimitChecker
-  attr_accessor :user
+  attr_accessor :user, :situation
```

Comment:
Btw, attr_writers are not used, so we can replace it with `attr_reader`. Or you can choose to use a setter like `self.situation = situation`